<launch>

  <arg name="use_urdf" default="false"/>

  <node name="mico_arm_driver" pkg="jaco_driver" type="jaco_arm_driver" output="screen" cwd="node">
    <!-- Set this parameter to use a specific arm on your system    --> 
    <!-- <param name="serial_number" value="PJ00650002132770005" /> -->
    <param name="tf_prefix" value="mico_" />

    <param name="fingers/stall_interval_seconds" value="1.0" />
    <param name="fingers/stall_threshold" value="1.0" />
    <param name="fingers/rate_hz" value="10.0" />
    <param name="fingers/tolerance" value="15.0" />

    <param name="arm_pose/stall_interval_seconds" value="1.0" />
    <param name="arm_pose/stall_threshold" value="0.005" />
    <param name="arm_pose/rate_hz" value="10.0" />
    <param name="arm_pose/tolerance" value="0.005" />
    <param name="arm_pose/tf_prefix" value="mico_" />

    <param name="joint_angles/stall_interval_seconds" value="1.0" />
    <param name="joint_angles/stall_threshold" value="1.0" />
    <param name="joint_angles/rate_hz" value="10.0" />
    <param name="joint_angles/tolerance" value="1.0" />
      
  </node>
    
  <group unless="$(arg use_urdf)">
    <node name="mico_tf_updater" pkg="jaco_driver" type="jaco_tf_updater" output="screen" cwd="node">
        <remap from="/mico_tf_updater/in/joint_angles" to="/mico_arm_driver/out/joint_angles"/>
        <param name="tf_prefix" value="mico_" />
        <param name="base_to_api" value="0.028" />
        <param name="base_to_j1" value="0.1544" />
        <param name="j1_to_j2" value="-0.1181" />
        <param name="j2_to_j3" value="0.2900" />
        <param name="j3_offset" value="-0.0070" />
        <param name="j3_to_j4" value="0.1233" />
        <param name="j4_to_j5" value="0.0741" />
        <param name="j5_to_j6" value="0.0741" />
        <param name="j6_to_end" value="0.1600" />
        <param name="j5_bend_degrees" value="-55.0" />
        <param name="j6_bend_degrees" value="55.0" />
    </node>
  </group>
  
  <group if="$(arg use_urdf)">
    <include file="$(find jaco_model)/launch/upload_mico.launch"/>
    <node name="mico_state_publisher"
           pkg="robot_state_publisher"
          type="robot_state_publisher">
      <remap from="joint_states" to="/mico_arm_driver/out/joint_state"/>
    </node>
  </group>

</launch>
